util_codes/View.py

```diff
@@ -0,0 +1,1128 @@
+            xytext=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax0ymin+(ax0ymax-ax0ymin)/4.), textcoords='data',
+            fontsize=14.)
+
+        ax1.annotate(("$\sigma$ =%0.5f S/m")%(self.sigvec[sigind]),
+            xy=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax1ymin+(ax1ymax-ax1ymin)/4.), xycoords='data',
+            xytext=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax1ymin+(ax1ymax-ax1ymin)/4.), textcoords='data',
+            fontsize=14.)
+
+        return ax0,ax1
+
+    def plot_1D_AP_f_x(self,absloc,coordloc,ax0,ax1,sigind):
+
+        if self.normal.upper() == "Z":
+            obsLoc=np.c_[absloc,coordloc,self.z]
+        elif self.normal.upper() == "Y":
+            obsLoc=np.c_[absloc,self.y,coordloc]
+        elif self.normal.upper() == "X":
+            obsLoc=np.c_[self.x,absloc,coordloc]
+
+        self.eval_loc(self.srcLoc,obsLoc, self.sigvec, self.fvec, self.orientation,self.func)
+        
+        ax0.set_xlabel("Frequency (Hz)")
+        ax1.set_xlabel("Frequency (Hz)")
+        ax0.set_ylabel("E field, Amplitude (V/m)")
+        ax1.set_ylabel("E field, Phase")
+            
+        #ax0.plot(self.sigvec[sigind]*np.ones_like(self.val_xfs.real[:,freqind]),
+        #         np.linspace(ax0ymin,ax0ymax,len(self.val_xfs.real[:,freqind])),linestyle="dashed",color="black",linewidth=3.0)
+
+        
+        ax0.plot(self.fvec,np.absolute(self.val_xfs[sigind,:]),color="blue")
+        ax1.plot(self.fvec,phase(self.val_xfs[sigind,:]),color="red")
+        
+        ax0ymin, ax0ymax = np.absolute(self.val_xfs[sigind,:]).min(),np.absolute(self.val_xfs[sigind,:]).max()
+        ax1ymin, ax1ymax = phase(self.val_xfs[sigind,:]).min(),phase(self.val_xfs[sigind,:]).max()
+
+        #ax2.plot(self.fvec[freqind]*np.ones_like(self.val_xfs[sigind,:]),
+        #            np.linspace(ax2ymin,ax2ymax,len(self.val_xfs[sigind,:])),linestyle="dashed",color="black",linewidth=3.0)
+        
+        ax0.set_ylim(ax0ymin, ax0ymax)
+        ax1.set_ylim(ax1ymin, ax1ymax)
+
+        ax0.set_xscale('log')
+        ax1.set_xscale('log')
+
+        ax0.annotate(("$\sigma$ =%0.5f S/m")%(self.sigvec[sigind]),
+            xy=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax0ymin+(ax0ymax-ax0ymin)/4.), xycoords='data',
+            xytext=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax0ymin+(ax0ymax-ax0ymin)/4.), textcoords='data',
+            fontsize=14.)
+
+        ax1.annotate(("$\sigma$ =%0.5f S/m")%(self.sigvec[sigind]),
+            xy=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax1ymin+(ax1ymax-ax1ymin)/4.), xycoords='data',
+            xytext=(10.**((np.log10(self.fvec.min())+np.log10(self.fvec.max()))/2), ax1ymin+(ax1ymax-ax1ymin)/4.), textcoords='data',
+            fontsize=14.)
+
+        return ax0,ax1
+
+    def plot_1D_RI_sig_x(self,absloc,coordloc,ax0,ax1,freqind):
```

Comment:
maybe rename "sig" to CC (unless there is stuff highly specific to sig here) since the necessary part is the fact that it lives in a cc - correct me if I am off here